butterfree/transform/aggregated_feature_set.py

```diff
@@ -543,18 +538,44 @@ def construct(
                 for w in self._windows
             ]
 
-            # left join each aggregation result to our base dataframe
-            output_df = reduce(
-                lambda left, right: self._dataframe_join(
-                    left,
-                    right,
-                    on=self.keys_columns + [self.timestamp_column],
-                    how="left",
-                    num_processors=num_processors,
-                ),
-                agg_list,
-                base_df,
-            )
+            # prepare our left table, a cartesian product between distinct keys
+            # and dates in range for this feature set
+
+            # todo next versions won't use this logic anymore,
+            # leaving for the client to correct the usage of aggregations
+            # without events
+
+            # keeping this logic to maintain the same behavior for already implemented
+            # feature sets
+
+            if self._windows[0].slide == "1 day":
+                base_df = self._get_base_dataframe(
+                    client=client, dataframe=output_df, end_date=end_date
+                )
+
+                # left join each aggregation result to our base dataframe
+                output_df = reduce(
+                    lambda left, right: self._dataframe_join(
+                        left,
+                        right,
+                        on=self.keys_columns + [self.timestamp_column],
+                        how="left",
+                        num_processors=num_processors,
+                    ),
+                    agg_list,
+                    base_df,
+                )
+            else:
```

Comment:
This is like our first implementation of this method, it'll be computationally expensive (due to the full outer join), but I don't see a better way of doing it for now. LGTM!